From be9f6fca63de19e90bfc9def077897a0dc7e3513 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 20 Feb 2023 13:23:02 +0100
Subject: [PATCH 089/841] feat(stm32mp1): manage STM32MP15 rev.Y

The new SoC version for STM32MP15 is the revision Y. The register
DBGMCU_IDC is updated for this new version with the value 0x2003.
The function stm32mp_get_soc_name() should also be updated to manage
this new SoC revision.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 7c11ab5c21362a0884bb68a96c98b7ce7dab3193 feat(stm32mp1): manage STM32MP15 rev.Y]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I32d6dd34f461aec37a6088f9f1341737d56b1658
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/291717
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 plat/st/stm32mp1/stm32mp1_def.h     | 1 +
 plat/st/stm32mp1/stm32mp1_private.c | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/plat/st/stm32mp1/stm32mp1_def.h b/plat/st/stm32mp1/stm32mp1_def.h
index 22a75eb2a..20c30be6d 100644
--- a/plat/st/stm32mp1/stm32mp1_def.h
+++ b/plat/st/stm32mp1/stm32mp1_def.h
@@ -74,6 +74,7 @@
 #define STM32MP1_REV_Z		U(0x1001)
 #endif
 #if STM32MP15
+#define STM32MP1_REV_Y		U(0x2003)
 #define STM32MP1_REV_Z		U(0x2001)
 #endif
 
diff --git a/plat/st/stm32mp1/stm32mp1_private.c b/plat/st/stm32mp1/stm32mp1_private.c
index 2dd88edcb..db40a156b 100644
--- a/plat/st/stm32mp1/stm32mp1_private.c
+++ b/plat/st/stm32mp1/stm32mp1_private.c
@@ -468,11 +468,9 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 	case STM32MP1_REV_B:
 		cpu_r = "B";
 		break;
-#if STM32MP13
 	case STM32MP1_REV_Y:
 		cpu_r = "Y";
 		break;
-#endif
 	case STM32MP1_REV_Z:
 		cpu_r = "Z";
 		break;
-- 
2.47.2

